modules/lnd.nix

```diff
@@ -0,0 +1,136 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+  
+let
+  nix-bitcoin-services = pkgs.callPackage ./nix-bitcoin-services.nix { };
+  cfg = config.services.lnd;
+  bitcoindDataDir = config.services.bitcoind.dataDir;
+  configFile = pkgs.writeText "lnd.conf" ''
+    [Application Options]
+    datadir=${cfg.dataDir}
+    bitcoin.mainnet=1
+    tlscertpath=/secrets/tls_cert
+    tlskeypath=/secrets/tls_key
+
+    [Bitcoin]
+    bitcoin.active=1
+    bitcoin.node=bitcoind
+
+    [tor]
+    tor.active=true
+    tor.v3=true
+    tor.streamisolation=true
+
+    [Bitcoind]
+    bitcoind.rpcuser=${cfg.bitcoin-rpcuser}
+    bitcoind.zmqpubrawblock=ipc://${bitcoindDataDir}/zmqpubrawblock.sock
+    bitcoind.zmqpubrawtx=ipc://${bitcoindDataDir}/zmqpubrawtx.sock
+  '';
+  init-lnd-wallet-script = pkgs.writeText "init-lnd-wallet.sh" ''
+#!/bin/sh
+
+set -e
+
+${pkgs.coreutils}/bin/sleep 5
+
+if [ ! -f ${cfg.dataDir}/chain/bitcoin/mainnet/wallet.db ]
+then
+  ${pkgs.coreutils}/bin/echo creating LND wallet
+
+  ${pkgs.curl}/bin/curl -s \
+  --cacert /secrets/tls_cert \
+  -X GET https://127.0.0.1:8080/v1/genseed | ${pkgs.jq}/bin/jq -c '.cipher_seed_mnemonic' > /secrets/lnd-seed-mnemonic
+
+  ${pkgs.curl}/bin/curl -s \
+  --cacert /secrets/tls_cert \
+  -X POST -d "{\"wallet_password\": \"$(${pkgs.coreutils}/bin/cat /secrets/lnd-wallet-password | ${pkgs.coreutils}/bin/tr -d '\n' | ${pkgs.coreutils}/bin/base64 -w0)\", \
+  \"cipher_seed_mnemonic\": $(${pkgs.coreutils}/bin/cat /secrets/lnd-seed-mnemonic | ${pkgs.coreutils}/bin/tr -d '\n')}" \
+  https://127.0.0.1:8080/v1/initwallet
+
+  ${pkgs.coreutils}/bin/echo wallet created
+else
+  ${pkgs.coreutils}/bin/echo unlocking wallet
+
+  ${pkgs.curl}/bin/curl -s \
+      -H "Grpc-Metadata-macaroon: $(${pkgs.xxd}/bin/xxd -ps -u -c 1000 ${cfg.dataDir}/chain/bitcoin/mainnet/admin.macaroon)" \
+      --cacert /secrets/tls_cert \
+      -X POST \
+      -d "{\"wallet_password\": \"$(${pkgs.coreutils}/bin/cat /secrets/lnd-wallet-password | ${pkgs.coreutils}/bin/tr -d '\n' | ${pkgs.coreutils}/bin/base64 -w0)\"}" \
+      https://127.0.0.1:8080/v1/unlockwallet
+  fi
+
+  ${pkgs.coreutils}/bin/echo wallet unlocked
+exit 0
+  '';
+in {
+
+  options.services.lnd = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        If enabled, the LND service will be installed.
+      '';
+    };
+    bitcoin-rpcuser = mkOption {
```

Comment:
Any particular reason why this is an option and `bitcoind.dataDir` isn't? Perhaps best to just use `config.services.bitcoind.dataDir` because then you can't forget to set the option, both in lnd and in bitcoind.

Reply:
Sorry, I don't understand this comment related to `bitcoin-rpcuser`. Lnd can use `bitcoin.conf` to get the rpc user, rpc password and zmq addresses, but I guess for this it would need read access to it, and that's what we want to remove, right? But you're probably talking about something else here and I'm completely missing the point?

Reply:
Seems like a misunderstaning indeed. You're getting the path of the bitcoin data directory by doing `bitcoindDataDir = config.services.bitcoind.dataDir;` in the let block. Similarly you could do `bitcoindRpcUser = config.services.bitcoind.rpcuser` and then you wouldn't have to have that option which people can forget to set.